runtime/wasmer.go

```diff
@@ -115,12 +119,27 @@ func (r *Runtime) Load(location, length int32) []byte {
 	return mem[location : location+length]
 }
 
-func (r *Runtime) Exec(function string, data, len int32) ([]byte, error) {
+func (r *Runtime) Exec(function string, loc int32, data interface{}) ([]byte, error) {
+	r.mutex.Lock()
+
+	// Find out the type of data & store it correctly into memory
+	var leng int32 = 0
+	switch data.(type) {
```

Comment:
not sure if this is needed, just passing in the length might be simpler than having to keep track of all the different possible types here